eZ/Publish/Core/Persistence/Cache/UrlWildcardHandler.php

```diff
@@ -0,0 +1,161 @@
+        );
+
+        $urlWildcard = $this->load($id);
+
+        $this->persistenceHandler->urlWildcardHandler()->remove($urlWildcard->id);
+
+        // need to clear lists of UrlWildcards cached due to loadAll()
+        $this->cache->invalidateTags(['ez-urlWildcard-all']);
+    }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::load
+     */
+    public function load($id)
+    {
+        $this->logger->logCall(
+            __METHOD__,
+            [
+                'id' => $id,
+            ]
+        );
+
+        $cacheItem = $this->cache->getItem('ez-urlWildcard-id-' . $id);
+
+        if ($cacheItem->isHit()) {
+            return $cacheItem->get();
+        }
+
+        $urlWildcard = $this->persistenceHandler->urlWildcardHandler()->load($id);
+
+        $cacheItem->set($urlWildcard);
+        $cacheItem->tag($this->getCacheTags([$urlWildcard]));
+        $this->cache->save($cacheItem);
+
+        $cacheItemBySource = $this->cache->getItem('ez-urlWildcard-source-' . $this->escapeForCacheKey($urlWildcard->sourceUrl));
+        $cacheItemBySource->set($urlWildcard);
+        $cacheItem->tag($this->getCacheTags([$urlWildcard]));
+        $this->cache->save($cacheItemBySource);
+
+        return $urlWildcard;
+    }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::loadAll
+     */
+    public function loadAll($offset = 0, $limit = -1)
+    {
+        $this->logger->logCall(
+            __METHOD__,
+            [
+                'offset' => $offset,
+                'limit' => $limit,
+            ]
+        );
+
+        $cacheItem = $this->cache->getItem('ez-urlWildcard-all-' . $offset . '-' . $limit);
+
+        if ($cacheItem->isHit()) {
+            return $cacheItem->get();
+        }
+
+        $urlWildcards = $this->persistenceHandler->urlWildcardHandler()->loadAll($offset, $limit);
+        $cacheItem->set($urlWildcards);
+        $cacheItem->tag($this->getCacheTags($urlWildcards));
+        $cacheItem->tag(
+            [
+                'ez-urlWildcard-all-' . $offset . '-' . $limit,
+            ]
+        );
+        $this->cache->save($cacheItem);
+
+        return $urlWildcards;
+    }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::lookup
+     */
+    public function lookup($url)
+    {
+        $this->logger->logCall(
+            __METHOD__,
+            [
+                'url' => $url
+            ]
+        );
+
+        $cacheItem = $this->cache->getItem('ez-urlWildcard-source-', $this->escapeForCacheKey($url));
+
+        if ($cacheItem->isHit()) {
+            return $cacheItem;
+        }
+
+        $urlWildcard = $this->persistenceHandler->urlWildcardHandler()->lookup($url);
+        $cacheItem->set($urlWildcard);
+        $cacheItem->tag($this->getCacheTags([$urlWildcard]));
+        $this->cache->save($cacheItem);
+
+        $cacheItemById = $this->cache->getItem('ez-urlWildcard-id-' . $urlWildcard->id);
```

Comment:
same, no need to pre warm things

Reply:
Alright, I understand, I've removed all pre-warmings here and there.